File: Toast/Sources/Toast/Helpers/RelativeDateFormat.swift
import Foundation

enum RelativeDateFormat {
    static func label(for date: Date, prefix: String) -> String {
        let interval = Date().timeIntervalSince(date)
        if interval < 60 {
            return "\(prefix) just now"
        }
        if interval < 3600 {
            let minutes = Int(interval / 60)
            return "\(prefix) \(minutes)m ago"
        }
        if interval < 86400 {
            let formatter = DateFormatter()
            formatter.timeStyle = .short
            formatter.dateStyle = .none
            return "\(prefix) \(formatter.string(from: date))"
        }
        let formatter = DateFormatter()
        formatter.dateStyle = .short
        formatter.timeStyle = .short
        return "\(prefix) \(formatter.string(from: date))"
    }
}

File: toast-win/src/Toast/Helpers/RelativeDateFormat.cs
namespace Toast.Helpers;

internal static class RelativeDateFormat
{
    public static string Label(DateTimeOffset date, string prefix)
    {
        var interval = DateTimeOffset.Now - date;
        if (interval.TotalSeconds < 60)
        {
            return $"{prefix} just now";
        }

        if (interval.TotalHours < 1)
        {
            var minutes = Math.Max(1, (int)interval.TotalMinutes);
            return $"{prefix} {minutes}m ago";
        }

        if (interval.TotalDays < 1)
        {
            return $"{prefix} {date.ToLocalTime():t}";
        }

        return $"{prefix} {date.ToLocalTime():g}";
    }

    public static DateTimeOffset? FromUnixMs(double? ms)
    {
        if (ms is null or <= 0)
        {
            return null;
        }

        return DateTimeOffset.FromUnixTimeMilliseconds((long)ms.Value);
    }
}
